packages/entando-apimanager/modules/api/apiManager.js

```diff
@@ -133,11 +133,37 @@ const getRequestParams = (request) => {
   return requestParams;
 };
 
+const isAbsoluteUrl = (url) => {
+  try {
+    new URL(url); // eslint-disable-line no-new
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
 const getCompleteRequestUrl = (request, params = {}) => {
+  // note on new URL():
+  // if url (first parameter) is an absolute path, base domain is not used
+  // if base domain is provided, it has to be absolute (even when url is absolute!)
+  // e.g.,
+  //  new URL('http://www.entando.com') -> { href: 'http://www.entando.com/', ... }
+  //  new URL('/digital-exchange', 'http://www.entando.com') -> { href: 'http://www.entando.com/digital-exchange', ... }
+  //  new URL('/digital-exchange') -> error [ERR_INVALID_URL]: Invalid URL: /digital-exchange
+  //  new URL('http://www.entando.com', 'not-url') -> error [ERR_INVALID_URL]: Invalid URL: not-url
+  //  new URL('http://www.entando.com', '/digital-exchange') -> error [ERR_INVALID_URL]: Invalid URL: /digital-exchange
+
   const storedDomain = getDomain(store.getState());
 
-  const domainUrl = new URL(request.domain || '', storedDomain);
-  const endpointUrl = new URL(domainUrl + request.uri);
+  const parsedStoredDomain = isAbsoluteUrl(storedDomain)
+    ? storedDomain
+    : new URL(new URL(window.location.href).origin + storedDomain).href;
```

Comment:
I'm almost sure it could be refactored in a simpler way. E.g. `new URL(window.location.href).origin` should be equivalent to `window.location.origin`